pkg/berglas/list.go

```diff
@@ -15,15 +15,52 @@
 package berglas
 
 import (
+	"cloud.google.com/go/storage"
 	"context"
 	"sort"
+	"time"
 
 	"github.com/pkg/errors"
 	"google.golang.org/api/iterator"
 )
 
+// Secret represents a specific secret stored in Google Cloud Storage
+// The attributes on this object should ideally map 1:1 with storage.ObjectAttrs
+type Secret struct {
+	// Name of the secret
+	Name string
+
+	// UpdatedAt indicates when a secret was last updated
+	UpdatedAt time.Time
+
+	// Generation indicates a secret's version
+	Generation int64
+}
+
+// SecretSlice is a list of secrets
+type SecretSlice []Secret
+
+// Len is the number of elements in the collection.
+func (s SecretSlice) Len() int {
+	return len(s)
+}
+
+// Less reports whether the element with
+// index i should sort before the element with index j.
+func (s SecretSlice) Less(i, j int) bool {
+	if s[i].Name == s[j].Name {
+		return s[i].Generation > s[j].Generation
+	}
+	return s[i].Name > s[j].Name
```

Comment:
I think this should be <, unless you want a reverse sort.
same with generation above.

Reply:
I do want a reverse sort. Newer generations are more relevant.